packages/vue/src/components/confirm-sign-in.vue

```diff
@@ -1,66 +1,61 @@
 <script setup lang="ts">
+import { computed, toRefs, useAttrs } from 'vue';
 import {
   authenticatorTextUtil,
-  getActorState,
   getFormDataFromEvent,
-  SignInState,
   translate,
 } from '@aws-amplify/ui';
-import { computed, ComputedRef, useAttrs } from 'vue';
 
-import { useAuth, useAuthenticator } from '../composables/useAuth';
+import { useAuthenticator } from '../composables/useAuth';
+import { UseAuthenticator } from '../types';
 import BaseFormFields from './primitives/base-form-fields.vue';
 
+/** @deprecated Component events are deprecated and not maintained. */
 const emit = defineEmits(['confirmSignInSubmit', 'backToSignInClicked']);
 const attrs = useAttrs();
 
-const { state, send } = useAuth();
+// `useAuthenticator` is casted for temporary type safety on this file.
+const props = useAuthenticator() as UseAuthenticator;
 
-const props = useAuthenticator();
+const { submitForm, toSignIn, updateForm } = props;
+const { user, error, isPending } = toRefs(props);
 
-const actorState = computed(() =>
-  getActorState(state.value)
-) as ComputedRef<SignInState>;
-const challengeName = actorState.value.context.challengeName;
+const challengeName = computed(() => user.value.challengeName);
 
 // Text Util
 const { getBackToSignInText, getConfirmText, getChallengeText } =
   authenticatorTextUtil;
 
 // Computed Properties
-const confirmSignInHeading = computed(() => getChallengeText(challengeName));
+const confirmSignInHeading = computed(() =>
+  getChallengeText(challengeName.value)
+);
 const backSignInText = computed(() => getBackToSignInText());
 const confirmText = computed(() => getConfirmText());
 
 // Methods
 const onInput = (e: Event): void => {
   const { name, value } = e.target as HTMLInputElement;
-  send({
-    type: 'CHANGE',
-    //@ts-ignore
-    data: { name, value },
-  });
+  updateForm({ name, value });
 };
 
 const onConfirmSignInSubmit = (e: Event): void => {
+  // TODO(BREAKING): remove unused emit
+  // istanbul ignore next
   if (attrs?.onConfirmSignInSubmit) {
     emit('confirmSignInSubmit', e);
   } else {
-    submit(e);
+    submitForm(getFormDataFromEvent(e));
   }
 };
 
-const submit = (e: Event): void => {
-  props.submitForm(getFormDataFromEvent(e));
-};
-
 const onBackToSignInClicked = (): void => {
+  // TODO(BREAKING): remove unused emit
+  // istanbul ignore next
   if (attrs?.onBackToSignInClicked) {
     emit('backToSignInClicked');
   } else {
-    send({
-      type: 'SIGN_IN',
-    });
+    toSignIn();
```

Comment:
I know this is out of scope for this PR, but is this method name not a little confusing? I'm definitely not that familiar with authenticator code but I almost feel like this name is missing a verb lol

Reply:
Yep, what it's really trying to say `navigateToSignInRoute`. This is part of public API, so we'll rely on next breaking changes to improve this.